lib/core/IncomingWebSocket.js

```diff
@@ -115,8 +115,10 @@ class IncomingWebSocket extends EventEmitter {
   /**
    * Closes the socket connection.
    */
-  close() {
+  close(code, msg) {
     this.teardown = true;
+    if(code >= 1000 && code < 1004)
```

Comment:
This can be risky let's say 1003 we don't want to close what will happen? & whats the thought process behind adding this check?

Reply:
This is within the close method, so the codes are basically a type of closure.
https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent/code
Basically, when the ws is open, the ws library allows `close` only for valid status codes and their check is 
```
function isValidStatusCode(code) {
  return (
    (code >= 1000 &&
      code <= 1014 &&
      code !== 1004 &&
      code !== 1005 &&
      code !== 1006) ||
    (code >= 3000 && code <= 4999)
  );
}
```
1004 is undefined and 1005 and 1006 are abrupt closures.
Basically the exception was happening as we were doing a terminate from hub side which sent a 1005/1006 and then were trying to close the CDP <-> client socket with that close, which threw the exception as it wasn't a valid close code.
Hence, we'll execute `close` method only when it's a valid status code and rest of the times, it'll be a terminate (which was happening earlier).

Reply:
Discussed with @Archish27